ml/surrogates.py

```diff
@@ -0,0 +1,56 @@
+from __future__ import print_function
+
+from sklearn.utils.testing import all_estimators
+from sklearn import base
+
+class sklearnRegressors(object):
+    """Scikit learn regressor factory.
+
+    Usage: 
+    import surrogates
+    rf1 = surrogates.sklearnRegressors.factory('RandomForestRegressor', n_estimators=5, max_depth=3)
+    gp1 = surrogates.sklearnRegressors.factory('GaussianProcessRegressor')
+    choices = surrogates.sklearnRegressors.names()
+    """
+
+    def reg_dict():
+        _all_regressors = {}
+        estimators = all_estimators()
+        for name, class_ in estimators:
+            if issubclass(class_, base.RegressorMixin):
+                _all_regressors[name] = class_
+        return _all_regressors
+
+    all_regs = reg_dict()
+
+    @classmethod
+    def factory(cls, name, *args, **kwargs):
+        if name in cls.all_regs:
+            return cls.all_regs[name](*args, **kwargs)
+        else:
+            print("Unknown regressor name "+name+"! For valid choices see sklearnRegressors.names()")
```

Comment:
I'd recommend raising an exception here instead of just printing, that way learn.py can fail with more helpful error messages instead of trying to do fitting with the wrong object when this else gets hit.

Reply:
good idea: will do